File: problem/Flip_Square_Vertical/reverseSubmatrix.py
class Solution:
    def reverseSubmatrix(self, grid, x, y, k):
        for i in range(k >> 1):
            for j in range(k):
                grid[x + i][y + j], grid[x + k - 1 - i][y + j] = grid[x + k - 1 - i][y + j], grid[x + i][y + j]
        
        return grid

File: problem/Flip_Square_Vertical/reverseSubmatrix.js
/**
 * @param {number[][]} grid
 * @param {number} x
 * @param {number} y
 * @param {number} k
 * @return {number[][]}
 */
const reverseSubmatrix = (grid, x, y, k) => {
    for (let i = 0; i < k >> 1; i++)
        for (let j = 0; j < k; j++)
            [grid[x + i][y + j], grid[x + k - 1 - i][y + j]] = 
                [grid[x + k - 1 - i][y + j], grid[x + i][y + j]];

    return grid;
};